[4.1] Remove Laravel 5.8 Support

tabacitu opened the pull request:
This should make it easier for us to maintain, and _maybe_ it'll allow Scrutinizer to actually install ```ocramius/package-versions``` 1.7.0 🤞 

Also, if we drop support for Laravel 5.8, we also drop support for PHP 7.1 which has reached EOL in January 2019, more than 1 year ago, so it's about time we stopped supporting it.

See:
- [Laravel Version Support](https://laravel.com/docs/7.x/releases#support-policy)
- [PHP Version Support](https://www.php.net/supported-versions.php)

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Laravel-Backpack/CRUD/inspections/6606c0b6-8794-4210-8f4c-68fa4f694cba) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Laravel-Backpack/CRUD/inspections/c6456315-5a65-449b-845c-be15216e3f64) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Laravel-Backpack/CRUD/inspections/ab7fed8e-e083-43cb-be50-f873be589aea) was created.

scrutinizer-notifier commented:
A [new inspection](https://scrutinizer-ci.com/g/Laravel-Backpack/CRUD/inspections/a1fa0290-5607-4ee7-8293-5654af7ad8c7) was created.

tabacitu commented:
Done. In the process I've also:
- fixed 2 places where we were still using array helpers despite having removed ```laravel/helpers```
- fixed Scrutinizer CI requiring wrong version of ```ocramius/package-versions```
- fixed Travis CI not sending code coverage status to Scrutinizer CI (hopefully 🤞 )

The result is that we FINALLY have ScrutinizerCI working for the 4.1 branch, once this is merged, so that we can see the difference in code quality between ```master``` and ```4.1```.

tabacitu commented:
PS. Scrutinizer's code rating increased from 9.1 to 9.5 , between ```master``` and ```4.1``` 🎉 🎈 🎊
